### DISCUSSION TITLE
How to get words with bbox and font attributes?

### DISCUSSION THREAD

**Discussion by JawaClass** [pymupdf/PyMuPDF#2335] [Q&A] [answered]
x0, y0, x1, y1, word, font, size, color, flag ...

I tried my luck with using the get_text('dict') output but couldnt get my x0 and x1 of the bbox correct.

Is there something implemented Im missing?

**Comment by JorjMcKie** [comment]
This a "Discussions" item, so first move it to there.

**Reply by JorjMcKie** [reply]
The "words" variant is highspeed and therefore only provides the bare mnimum of information.
In addition, think about "***this***word"! In the real world it would consist of 2 different fonts, wouldn't it?
So, by definition of what a "word" is, font characteristics **_cannot_** be assciated with it ... neither of course any of the other font properties or the color.

Therefore, the bbox is all you can have for a word - plus some technical information like block, line and word number.

**Reply by JorjMcKie** [reply]
The "dict" output by definition separates a line in 1 or more text pieces. One piece (= "span") exactly in case all characters in the line have the same baseline and the same font, font attributes and color.

**Reply by JawaClass** [reply]
I am aware of this limitation but in the documents I have to work with such words with multiple font styles mixed are super uncommon if not non existent. But this information would be nice to have. In such cases I would just take the most common font style within a word. Is building the bbox from "dict" adviced or via "rawdict".

**Reply by JorjMcKie** [reply]
If you want to rely on that all characters of your words will occur in the same span, you best should use "rawdict".
You can also make a list of all spans and then, for a given word, find the span bbox that contains the word's bbox. Might even be the faster alternative.

**Reply by JorjMcKie** [reply, marked as answer]
> "super uncommon if not non existent"

I doubt that very much: "**bold**." and "**bold**-*italic*" are returned as words with one single bbox. The first word has a non-bold `.` appended which therefore will not be in the same span as the first part. And constructs like "**N**orth **A**tlantic **T**reaty **O**rganisation" are by no means uncommon.
